cuda_core/cuda/core/experimental/_module.py

```diff
@@ -46,26 +73,27 @@ def __init__(self, module, code_type, jit_options=None, *,
                  symbol_mapping=None):
         if code_type not in self._supported_code_type:
             raise ValueError
+        _lazy_init()
         self._handle = None
 
-        driver_ver = handle_return(cuda.cuDriverGetVersion())
-        self._loader = _backend["new"] if driver_ver >= 12000 else _backend["old"]
+        backend = "new" if (_py_major_ver >= 12 and _driver_ver >= 12000) else "old"
```

Comment:
I find that the checks done within `ObjectCode.__init__` are mixing styles between using `_driver_ver` and `backend`. They serve the same purpose, perhaps switching to only `_driver_ver` would be better? Then simplifying our global `_backend` to an empty dict such that it gets filled out by `_lazy_init()` as appropriate. (I'm taking some inspiration from the _launcher.py change.)

Reply:
Yeah I updated the PR description to cover this. We need to ensure both 1. the bindings are available (this needs `_py_major_ver` check) and 2. the driver supports it (this needs `_driver_ver` check). I think `_launcher.py` is doing a similar check too?

Reply:
(Not sure if this would be better: In the future we might need an internal feature map that can be referenced across modules, so that we can just detect versions in one place.)

Reply:
Ah my bad, I overlooked the `_py_major_ver`.

I what I find still awkward is that the global `_backend` is a dict of "old" and "new". But only one of these is ever valid and we know which one that is after `_lazy_init()` is called. So starting out with an empty dict that gets populated by the `_lazy_init()` is what I'd prefer.

Reply:
This is a bit trickier than I thought, because we also need the dict key "old"/"new" as a proxy to prepare for `args` (which is different for `cuModuleLoadDataEx`/`cuLibraryLoadData`). Let's improve this after beta 1.

Reply:
Sounds good.

Reply:
Tracked in https://github.com/NVIDIA/cuda-python/issues/193.